lib/github.js

```diff
@@ -1,20 +1,246 @@
+  const variables = {
+    login,
+    repoName,
+    first: 100
+  }
+
+  const {
+    errors,
+    data: {
+      repository: {
+        refs: {
+          nodes: heads
+        }
+      }
+    }
+  } = await createGithubFetchForUser(user)({ query: headsQuery, variables })
+  if (errors) {
+    throw new Error(JSON.stringify(errors))
+  }
+
+  return heads
+}
+
+const tag = async (
+  user,
+  repoId,
+  name,
+  message,
+  commitId
+) => {
+  const response = await fetch(`https://api.github.com/repos/${repoId}/git/tags`, {
+    method: 'POST',
+    headers: {
+      'Accept': 'application/vnd.github.v3+json',
+      'Authorization': `Bearer ${user.githubAccessToken}`
+    },
+    body: JSON.stringify({
+      tag: name,
+      message,
+      object: commitId,
+      type: 'commit',
+      tagger: {
+        name: user.email, // TODO
+        email: user.email,
+        date: new Date()
+      }
+    })
+  })
+  return response.json()
+}
+
+const createRef = async (
+  token,
+  repoId,
+  ref,
+  sha
+) => {
+  const response = await fetch(`https://api.github.com/repos/${repoId}/git/refs`, {
+    method: 'POST',
+    headers: {
+      'Accept': 'application/vnd.github.v3+json',
+      'Authorization': `Bearer ${token}`
+    },
+    body: JSON.stringify({
+      ref,
+      sha
     })
+  })
+  return response.json()
+}
+
+const removeRef = async (
+  token,
+  repoId,
+  ref
+) => {
+  const response = await fetch(`https://api.github.com/repos/${repoId}/git/refs/${ref}`, {
+    method: 'DELETE',
+    headers: {
+      'Accept': 'application/vnd.github.v3+json',
+      'Authorization': `Bearer ${token}`
+    }
+  })
+  return response.status === 204
+}
+
+const tags = async (user, repoId) => {
+  const tagsQuery = `
+    query repository(
+      $login: String!,
+      $repoName: String!,
+      $first: Int
+    ) {
+      repository(owner: $login, name: $repoName) {
+        refs(refPrefix: "refs/tags/", first: $first) {
+          nodes {
+            name
+            target {
+              ... on Tag {
+                name
+                message
+                oid
+                author: tagger {
+                  name
+                  email
+                  date
+                }
+                commit: target {
+                  ... on Commit {
+                    id: oid
+                  }
+                }
+              }
+            }
+          }
+        }
+      }
+    }
+  `
+  const [login, repoName] = repoId.split('/')
+  const variables = {
+    login,
+    repoName,
+    first: 100
+  }
 
-    return response.json()
+  const {
+    errors,
+    data: {
+      repository: {
+        refs: {
+          nodes: tags
+        }
+      }
+    }
+  } = await createGithubFetchForUser(user)({ query: tagsQuery, variables })
+  if (errors) {
+    throw new Error(JSON.stringify(errors))
   }
+
+  return tags
+    .map(tag => tag.target)
+    .map(tag => ({
+      ...tag,
```

Comment:
Throws an error for me:
/haku-backend/lib/github.js:200
      ...tag,
      ^^^
SyntaxError: Unexpected token ...
    at createScript (vm.js:74:10)
    at Object.runInThisContext (vm.js:116:10)
    at Module._compile (module.js:533:28)
    at Object.Module._extensions..js (module.js:580:10)
    at Module.load (module.js:503:32)
    at tryModuleLoad (module.js:466:12)
    at Function.Module._load (module.js:458:3)
    at Module.require (module.js:513:17)
    at require (internal/module.js:11:18)
    at Object.<anonymous> (/Users/daniel/work/haku-backend/graphql/resolvers/Commit.js:1:100)
    at Module._compile (module.js:569:30)
    at Object.Module._extensions..js (module.js:580:10)
    at Module.load (module.js:503:32)
    at tryModuleLoad (module.js:466:12)
    at Function.Module._load (module.js:458:3)
    at Module.require (module.js:513:17)